src/electron/platform/android/setup/live-android-setup-deps.ts

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { UserConfigurationStore } from 'background/stores/global/user-configuration-store';
+import { compareSemverValues, SemverComparisonResult } from 'electron/common/semver-comparer';
+import { AndroidServiceApkLocator } from 'electron/platform/android/android-service-apk-locator';
+import {
+    AndroidServiceConfigurator,
+    AndroidServiceConfiguratorFactory,
+    DeviceInfo,
+    PackageInfo,
+    PermissionInfo,
+} from 'electron/platform/android/android-service-configurator';
+import { AndroidSetupDeps } from 'electron/platform/android/setup/android-setup-deps';
+
+export class LiveAndroidSetupDeps implements AndroidSetupDeps {
+    private serviceConfig: AndroidServiceConfigurator;
+    private selectedDeviceId: string;
+
+    constructor(
+        private readonly configFactory: AndroidServiceConfiguratorFactory,
+        private readonly configStore: UserConfigurationStore,
+        private readonly apkLocator: AndroidServiceApkLocator,
+    ) {}
+
+    public hasAdbPath = async (): Promise<boolean> => {
+        try {
+            const adbLocation = this.configStore.getState().adbLocation;
+            this.serviceConfig = await this.configFactory.getServiceConfigurator(adbLocation);
+            return true;
+        } catch (error) {
+            return false;
+        }
+    };
+
+    public setAdbPath = (path: string): void => {
+        this.configStore.getState().adbLocation = path;
+    };
+
+    public getDevices = async (): Promise<DeviceInfo[]> => {
+        return await this.serviceConfig.getConnectedDevices();
+    };
+
+    public setSelectedDeviceId = (id: string): void => {
+        this.selectedDeviceId = id;
+    };
+
+    public hasExpectedServiceVersion = async (): Promise<boolean> => {
+        try {
+            const installedVersion: string = await this.getInstalledVersion();
+            if (installedVersion) {
+                return installedVersion === (await this.getTargetVersion());
+            }
+        } catch (error) {
+            console.log(error);
```

Comment:
This tends to clutter up the test. Is there a recommended way to just silently eat an exception? When I remove the console.log call, I get warnings about having an empty block. Not quite sure how to handle this...

Reply:
The usual way we handle that is to inject a `Logger` dependency rather than invoking `console` directly, that way you can just stub it as a no-op logger for the error cases

Reply:
Sweet! Will do